### PR TITLE
docs: fix critical security documentation errors (Issue #282)

### LINKED ISSUE #282: Docs: Audit and clean up existing documentation
## Goal

Before writing new documentation, audit and clean up all existing docs to ensure they are a solid foundation.

## Philosophy

The v0.1.0 documentation should be a **foundational kernel** - concise, accurate, and true. Quality over quantity.

**What we want:**
- Factually correct and verified against current code
- Concise - says what's needed, nothing more
- Well-written and clear
- Consistent tone and formatting

**What we DON'T want:**
- Boilerplate or filler text
- Expansive prose or lengthy examples
- Discussion of development process or future work
- Speculative features or TODO placeholders
- Redundant information across files

## Scope

Review all existing markdown files in the repo:
- README.md
- docs/ directory
- Any other .md files

For each file:
1. Verify all claims against actual code behavior
2. Remove or flag anything inaccurate
3. Trim unnecessary verbosity
4. Remove future work / roadmap / TODO sections
5. Ensure formatting is consistent

## Acceptance Criteria

- [ ] All existing docs reviewed
- [ ] Inaccuracies corrected or removed
- [ ] Verbose/unnecessary content trimmed
- [ ] No speculative or future-facing content remains
- [ ] Consistent formatting throughout

---
(AI-generated via Claude Code w/ Opus 4.5)

### PR DESCRIPTION
**Pull request by duplico**
## Summary

This PR fixes multiple critical security documentation errors in Magpie where artifact downloads were incorrectly documented as "public by default". In reality, artifact downloads require Bearer token authentication except for files explicitly placed in `/artifacts/public/*`.

### Critical Security Documentation Errors

1. **docs/user-guide.md** - Fixed API endpoint table
   - Corrected: `GET /artifacts/{path}/{tag}` and `GET /artifacts/{path}/blobs/{hash}` now correctly documented as requiring Bearer token or IP allow-list
   - Previously incorrectly listed as public endpoints
   - Added `/api/v1/auth/validate` to public endpoints (intentionally public for Caddy forward_auth)

2. **docs/ansible-integration.md** - Fixed multiple false claims about public downloads
   - Overview section: Downloads now correctly require Bearer token
   - Prerequisites section: Clarified tokens needed for all operations
   - Troubleshooting section: Fixed claim that 401/403 "should not occur" for downloads
   - Security best practices: Updated to reflect token requirement

3. **docs/authentik-setup.md** - Fixed misleading deployment defaults
   - Step 1: Clarified Bearer token authentication is the PRODUCTION DEFAULT (not public)
   - Step 2: Corrected instructions for enabling Authentik (replaces Bearer auth, not enables alongside public access)

4. **docs/authentik-testing-guide.md** - Fixed incorrect endpoint classification
   - Corrected comment claiming `/api/v1/artifacts` is a "public endpoint"

## Verification

All corrections are based on:
- Caddyfile.prod (commit 92f4ba2 added artifact download protection)
- Fact inventory generated from actual code review
- Protected routes explicitly use `@require_auth` matcher on all download endpoints

## Testing

- [x] Verified against Caddyfile.prod to ensure documentation accuracy
- [x] All changes are documentation-only (no code changes)
- [x] Links and references verified
- [x] No new documentation gaps introduced

Generated with Claude Code w/ Opus 4.5

Closes #282

### CODE REVIEW

File: docs/authentik-setup.md [hunk 1]
```diff
@@ -315,18 +315,24 @@ For fine-grained access control (e.g., restrict certain artifact paths), impleme
...
 
-### Step 2: Enable Authentik SSO
+### Step 2: Enable Authentik SSO (Replaces Bearer Tokens for Browser Access)
+
+To replace Bearer token authentication with Authentik SSO for browser access:
 
 1. Set `AUTHENTIK_HOST` environment variable
-2. Comment out the default handler and uncomment the Authentik SSO handler in Caddyfile.prod
+2. In Caddyfile.prod, replace the current Bearer token forward_auth block (lines 276-278)
+   with the Authentik forward_auth block (commented at lines 267-274)
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/authentik-setup.md]: Fixed in be860d5. Updated line references to match actual Caddyfile.prod.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/user-guide.md [hunk 2]
```diff
@@ -529,205 +389,73 @@ magpie gc
...
-|----------|-------|-------------|
-| `MAGPIE_SERVER` | Client | Server URL |
-| `MAGPIE_TOKEN` | Client | Auth token |
-| `MAGPIE_TIMEOUT` | Client | Request timeout (CLI/env only, not config file) |
-| `MAGPIE_HTTP_PORT` | Deploy | External HTTP port (default: 8080) |
-| `MAGPIE_HTTPS_PORT` | Deploy | External HTTPS port (default: 8443) |
-| `MAGPIE_STORAGE_PATH` | Server | Storage directory |
-| `MAGPIE_RETENTION_DAYS` | Server | GC retention period |
-| `MAGPIE_DEBUG` | Both | Enable debug mode |
+- `GET /health` - Health check
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/user-guide.md:431]: Fixed in be860d5. Added GET /api/v1/auth/validate to public endpoints section.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/user-guide.md [hunk 3]
```diff
@@ -529,205 +389,73 @@ magpie gc
...
-| DELETE | `/api/v1/tokens/{name}` | admin | Revoke token |
-| POST | `/api/v1/gc` | admin | Run GC |
-| POST | `/api/v1/tags/{tag_name}/flush` | admin | Flush tag globally |
+- `GET /api/v1/artifacts` - List artifacts (read)
+- `GET /api/v1/artifacts/{path}` - List versions (read)
+- `POST /api/v1/upload/{path}` - Upload (write)
+- `POST /api/v1/artifacts/{path}/{ref}/tags` - Create tag (write)
+- `DELETE /api/v1/artifacts/{path}/tags/{tag}` - Remove tag (write)
+- `POST /api/v1/gc` - Run GC (admin)
+- `POST /api/v1/tokens` - Create token (admin)
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/user-guide.md:445]: Fixed in be860d5. Added GET /artifacts/{path}/{tag} and GET /artifacts/{path}/blobs/{hash} to protected endpoints section.

(AI-generated via Claude Code w/ Opus 4.5)

[the remark this answers is not in the document]
Comment (duplico) [reply docs/user-guide.md:445]: Fixed in be860d5. Added GET /artifacts/{path}/{tag} and GET /artifacts/{path}/blobs/{hash} to protected endpoints section as noted.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/ansible-integration.md [hunk 4]
```diff
@@ -1,393 +1,185 @@
...
-         Authorization: "Bearer {{ magpie_token }}"
-       # ... upload configuration
-     no_log: true
-   ```
-4. **Pin versions for production** - Use specific tags or hash refs instead of `latest`
-5. **Verify integrity** - For security-sensitive artifacts, use one of these approaches:
-   - **Hash refs** (recommended): Download via `/blobs/<hash>` for content-addressed immutability
-   - **Pre-defined checksums**: Store expected SHA-256 in vault-encrypted variables
-6. **Rotate tokens periodically** - If using tokens, update vault-stored tokens on a schedule
+- **Use minimal token scope**: read-only for downloads, write for uploads, admin only for administration
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/ansible-integration.md]: Fixed in be860d5. Updated to clarify that read scope (or admin/write) is required for downloads, except /artifacts/public/* which needs no token.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/user-guide.md [hunk 5]
```diff
@@ -529,205 +389,73 @@ magpie gc
...
-  "https://magpie.example.com/api/v1/tags/old-release/flush?confirm_walk_filesystem=true&dry_run=true"
+# Download by tag
+curl -O https://magpie.example.com/artifacts/images/ubuntu/latest
 
-# Actually flush the tag
-curl -X POST \
-  -H "Authorization: Bearer mgp_admin_token" \
-  "https://magpie.example.com/api/v1/tags/old-release/flush?confirm_walk_filesystem=true"
+# Download by hash
+curl -O https://magpie.example.com/artifacts/images/ubuntu/blobs/a1b2c3d4
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/user-guide.md]: Fixed in be860d5. Added Authorization: Bearer headers to both curl download examples.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/ansible-integration.md [hunk 6]
```diff
@@ -1,393 +1,185 @@
...
 ```
 
-## Secure Token Storage (For Write Operations)
-
-Tokens are only required for write operations (uploads, tag management). If you
-only need to download artifacts, you can skip this section.
+## Token Storage (Write Operations Only)
 
-**Never store tokens in plaintext in playbooks or variable files.**
+Tokens required only for uploads and tag management. Download-only operations don't need tokens (or use public artifacts).
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/ansible-integration.md]: Fixed in be860d5. Corrected statement to clarify that tokens are required for downloads (except /artifacts/public/*), not just write operations.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/ansible-integration.md [hunk 7]
```diff
@@ -1,393 +1,185 @@
...
 
 ```yaml
-- name: Download artifact by hash ref
-  ansible.builtin.get_url:
-    url: "{{ magpie_server }}/artifacts/tools/validator/blobs/a1b2c3d4"
+- ansible.builtin.get_url:
+    url: "https://magpie.example.com/artifacts/tools/validator/blobs/a1b2c3d4"
     dest: /usr/local/bin/validator
     mode: "0755"
 ```
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/ansible-integration.md:44]: Fixed in be860d5. Added Authorization headers with Bearer token to all download examples.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/ansible-integration.md [hunk 8]
```diff
@@ -1,393 +1,185 @@
...
       - magpie_artifact_path | length > 0
       - magpie_dest | length > 0
-    fail_msg: "Required magpie_artifact variables not set"
 
-- name: Download artifact from Magpie
-  ansible.builtin.get_url:
+- ansible.builtin.get_url:
     url: "{{ magpie_server }}/artifacts/{{ magpie_artifact_path }}/{{ magpie_artifact_tag }}"
     dest: "{{ magpie_dest }}"
     mode: "{{ magpie_mode }}"
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/ansible-integration.md:156]: Fixed in be860d5. Added Authorization header to reusable role's get_url task and added magpie_token variable to role defaults.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/authentik-testing-guide.md [hunk 9]
```diff
@@ -1,369 +1,80 @@
...
-- AUTHENTIK_HOST: [set/unset]
-```
+**Invalid headers**: Verify forward_auth URL ends with `/auth/caddy`. Check provider type is "Forward auth (single application)".
 
-Include any errors, unexpected behavior, or edge cases encountered.
+**Bearer tokens broken**: Test `/api/v1/auth/validate`. Ensure API endpoints use Magpie's forward_auth, not Authentik's. Check Caddyfile route order.
 
----
-*This documentation was generated with AI assistance (Claude Code w/ Opus 4.5).*
+**Artifacts require auth**: Verify forward_auth block in `/artifacts/*` is commented out or AUTHENTIK_HOST is unset. Restart Caddy after changes.
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/authentik-testing-guide.md]: Fixed in be860d5. Corrected troubleshooting guidance to explain Bearer token auth is the default, not public access.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/ansible-integration.md [hunk 10]
```diff
@@ -1,393 +1,185 @@
...
 ```yaml
-- name: Download large artifact
-  ansible.builtin.get_url:
-    url: "{{ magpie_server }}/artifacts/images/large-vm/latest"
+- ansible.builtin.get_url:
+    url: "https://magpie.example.com/artifacts/images/large-vm/latest"
     dest: /var/lib/images/vm.qcow2
-    timeout: 600  # 10 minutes
-    mode: "0644"
+    timeout: 600
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/ansible-integration.md:186]: Fixed in be860d5. Added Authorization header with Bearer token to the timeout example.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/authentik-setup.md [hunk 11]
```diff
@@ -145,205 +133,40 @@ docker compose -f docker-compose.prod.yml restart caddy
...
-  -F "file=@test.tar.gz" \
-  https://magpie.example.com/api/v1/upload/images/test
-
-# Download via API with token (should work)
-magpie get images/test:latest
-```
+**Unauthenticated access**: Open incognito browser to `https://magpie.example.com/artifacts/` - should redirect to Authentik login.
 
-### Test 4: Session Persistence
+**Authenticated access**: Log in with Authentik credentials - should show directory listing.
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/authentik-setup.md]: Fixed in be860d5. Clarified testing section to explain that Bearer token auth is the default (401 Unauthorized), and Authentik SSO is optional.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/authentik-setup.md [hunk 12]
```diff
@@ -145,205 +133,40 @@ docker compose -f docker-compose.prod.yml restart caddy
...
 
-To disable SSO, comment out the forward_auth block and restart Caddy.
+## Authentication Methods
 
-## References
+| Method | Endpoint | Use Case |
+| --- | --- | --- |
+| Authentik SSO | `/artifacts/*` | Humans browsing artifacts |
+| Bearer tokens | `/api/v1/*` | CI/CD, scripts, automation |
+| Public (no auth) | `/artifacts/public/*` | Public artifacts |
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/authentik-setup.md]: Fixed in be860d5. Updated authentication methods table to show Bearer tokens as the production default for /artifacts/*, with Authentik SSO as optional replacement.

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/ansible-integration.md [hunk 13]
```diff
@@ -1,393 +1,185 @@
...
 
 ## Prerequisites
 
-- Magpie server accessible from Ansible control node or target hosts
-- (Optional) Magpie bearer token if your playbooks will upload artifacts or manage tags
-- (Recommended) Ansible Vault configured for secret storage when using bearer tokens
-
-## Basic Download Pattern
+- Magpie server accessible from Ansible control node
+- Bearer token (required for downloads; optional if using `/artifacts/public/*`)
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/ansible-integration.md]: Fixed in be860d5. Removed contradictory statement about downloads not needing tokens. Prerequisites now correctly state tokens are required for artifact operations (except /artifacts/public/*).

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/ansible-integration.md [hunk 14]
```diff
@@ -1,393 +1,185 @@
...
-### Role Structure
-
-```
-roles/magpie_artifact/
-  defaults/main.yml
-  tasks/main.yml
+- ansible.builtin.systemd:
+    name: myapp
+    state: restarted
+  when: download_result.changed
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/ansible-integration.md:129]: Fixed in be860d5. Added Authorization headers with Bearer token to all Common Patterns examples (deploy scripts, archives, restart service).

(AI-generated via Claude Code w/ Opus 4.5)

File: docs/authentik-testing-guide.md [hunk 15]
```diff
@@ -1,369 +1,80 @@
...
-**Note**: This test only applies if `AUTHENTIK_HOST` is configured and the forward_auth block is uncommented in Caddyfile.prod.
-
-### Steps
-1. Open an incognito/private browser window
-2. Navigate to `https://magpie.example.com/artifacts/`
-3. Observe behavior
+## Test 2: Browser Access (Unauthenticated)
 
-### Expected Results
+Open incognito browser to `https://magpie.example.com/artifacts/`. If Authentik SSO enabled, redirects to login page. Otherwise, shows directory listing.
```
[the remark this answers is not in the document]
Comment (duplico) [reply docs/authentik-testing-guide.md]: Fixed in be860d5. Corrected Test 2 to clarify that without Authentik SSO (default), browser access returns 401 Unauthorized, not directory listing.

(AI-generated via Claude Code w/ Opus 4.5)
